dashboard/controller/users/exportCSV.js

```diff
@@ -173,10 +178,34 @@ module.exports = function exportCSV(req, res) {
 				});
 		}
 	], function() {
-		if (users.length == 0) {
+		var filteredUsers = [];
+		var requiredIDs = new Set();
+		if(selectedClassroom !== 'undefined'){
+			selectedClassroom.split(',').forEach(id => {
+				requiredIDs.add(id);
+			});
+		}
+		users.forEach(user => {
+			let required = true;
+			if (selectedUsername !== 'undefined') {
+				if (user.name !== selectedUsername) {
+				  required = false;
+				}
+			}
+			if (required && selectedRole !== 'all' && user.type !== selectedRole) {
+				required = false;
+			}
+			if (required && selectedRole === 'student' && selectedClassroom !== 'undefined') {
+				if (!requiredIDs.has(user._id)) {
+				  required = false;
+				}
+			}
+			if(required) filteredUsers.push(user);
+		})
```

Comment:
- Filtering logic can be simplified.
- Some variable names are misleading, like `requiredIDs`.
- Fix indentation.
- Maybe consider using [Array.filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter).

Reply:
Okay I will look into it. Thank you for the feedback!